### PR TITLE
PS-1785 om tags

### PR DESCRIPTION
**Pull request by pivnicek**

### CODE REVIEW

**Review by odinuv — COMMENTED** [review]
two things:
- wasn't this supposed to include marking the file as permanant? (if not where is it)
- this was supposed to be applied only to specified components https://keboola.atlassian.net/wiki/spaces/TECH/pages/1116012545/New+File+Mapping#N%C3%A1vrh

**Comment by pivnicek** [comment]
Not as far as I knew.

marking permanent is for the `skip table creation option`.

that is the thing that is only for specific components too AFAIK

**Comment by pivnicek** [comment]
shit, you're right about the specific components though according to that document.  Also in the document `runID` is in the metadata.

I'll update this.

**Comment by odinuv** [comment]
This https://keboola.atlassian.net/browse/PS-1758 says it should be permanent (though this is not really related to this component)

**Comment by pivnicek** [comment]
Not entirely sure why it should only be for certain components, but ok, I'll update it.

**Comment by pivnicek** [comment]
yeah, but I wrote that so it's probably wrong :-)

**Comment by odinuv** [comment]
Well, if we release it as it is then files coming from transformations will start receiving these tags - which might or might not be desirable (i'm not objecting either solution, i'm just asking)

**Comment by pivnicek** [comment]
yeah, I don't know,  I don't think adding tags can hurt any workflow, it's non-destructive unless some script is using a tag count.

**Comment by pivnicek** [comment]
let's think about it for a bit, there's no immediate need to merge this right now.

**Comment by pivnicek** [comment]
btw, the output-mapping v14 requires at minimum componentId

**Comment by odinuv** [comment]
> btw, the output-mapping v14 requires at minimum componentId, so if it's component restricted will need to update the OM to remove that restriction.

or add an option not to write the metadata

**Comment by pivnicek** [comment]
uf.  Thanks

**Comment by pivnicek** [comment]
@odinuv  could you please take a look and let me know if this is debyl or not?  If it's ok then I'll continue with the update.

File: src/Docker/Runner/DataLoader/DataLoader.php [hunk 1]
```diff
@@ -256,6 +267,11 @@ public function storeOutput()
         }
     }
 
+    private function useFileMetadataTags()
+    {
+        return $this->component->allowUseFileStorageOnly() && $this->runtimeConfig['use-file-storage-only'];
```
Comment (odinuv) [inline src/Docker/Runner/DataLoader/DataLoader.php:272]: use_file_storage_only https://keboola.atlassian.net/browse/PS-1759

File: src/Docker/Runner.php [hunk 2]
```diff
@@ -240,6 +240,7 @@ private function runRow(JobDefinition $jobDefinition, $action, $mode, $jobId, Us
                 $this->loggersService->getLog(),
                 $workingDirectory->getDataDir(),
                 $configData['storage'],
+                $configData['runtime'],
```
Comment (odinuv) [inline src/Docker/Runner.php:243]: I'd put in the entire $jobDefinition, the refactoring will be easier and the interface cleaner (in future when the redundant fields are removed)

**Comment by pivnicek** [comment]
ok @odinuv the tests passed last time so feel free to have a look whenever:) (I've added one for this to the backend-specific tests since the runner tests are the bottleneck right now I think that should be ok)

**Review by odinuv — APPROVED** [review]

**Pull request merged**